manager/allocator/network.go

```diff
@@ -641,6 +644,35 @@ func (a *Allocator) commitAllocatedNode(ctx context.Context, batch *store.Batch,
 	return nil
 }
 
+func updatePortsInHostPublishMode(s *api.Service) {
```

Comment:
I think this function should be a method of `api.Service` given it is manipulating its member